astropy/coordinates/tests/test_representation_arithmetic.py

```diff
@@ -0,0 +1,322 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+# TEST_UNICODE_LITERALS
+import functools
+import numpy as np
+
+from ... import units as u
+from .. import (PhysicsSphericalRepresentation, CartesianRepresentation,
+                CylindricalRepresentation, SphericalRepresentation,
+                UnitSphericalRepresentation, Longitude, Latitude)
+from ..angle_utilities import angular_separation
+from ...tests.helper import pytest, assert_quantity_allclose
+
+
+def assert_representation_allclose(actual, desired, rtol=1.e-7, atol=None,
+                                   **kwargs):
+    assert_quantity_allclose(actual.to_cartesian().xyz,
+                             desired.to_cartesian().xyz,
+                             rtol, atol, **kwargs)
+
+
+def representation_equal(first, second):
+    return functools.reduce(np.logical_and,
+                            (getattr(first, component) ==
+                             getattr(second, component)
+                             for component in first.components))
+
+class TestArithmetic():
+    """Test absolute value, multiplication, and division."""
+
+    def setup(self):
+        self.lon = Longitude(np.arange(0, 12.1, 2), u.hourangle)
+        self.lat = Latitude(np.arange(-90, 91, 30), u.deg)
+        self.distance = [5., 12., 4., 2., 4., 12., 5.] * u.kpc
+        self.spherical = SphericalRepresentation(self.lon, self.lat,
+                                                 self.distance)
+        self.unit_spherical = self.spherical.represent_as(
+            UnitSphericalRepresentation)
+        self.cartesian = self.spherical.to_cartesian()
+
+    def test_abs_spherical(self):
+        abs_s = abs(self.spherical)
+        assert isinstance(abs_s, u.Quantity)
+        assert np.all(abs_s == self.distance)
+
+    @pytest.mark.parametrize('representation',
+                             (PhysicsSphericalRepresentation,
+                              CartesianRepresentation,
+                              CylindricalRepresentation))
+    def test_abs(self, representation):
+        in_rep = self.spherical.represent_as(representation)
+        abs_rep = abs(in_rep)
+        assert isinstance(abs_rep, u.Quantity)
+        assert_quantity_allclose(abs_rep, self.distance)
+
+    def test_abs_unitspherical(self):
+        abs_rep = abs(self.unit_spherical)
+        assert abs_rep.unit == u.dimensionless_unscaled
+        assert np.all(abs_rep == 1. * u.dimensionless_unscaled)
```

Comment:
For unit spherical, the length of the vectors should be identically unity, not approximately.